public/js/routing/router.js

```diff
@@ -15,10 +15,12 @@
 
 /* eslint-env browser */
 
-export default class Router {
-  constructor(window, shell, container) {
+import EventTarget from 'event-target-shim/dist/event-target-shim.min.js';
+
+export default class Router extends EventTarget {
```

Comment:
I agree that's an interesting choice. The decision was indeed to keep compatible with the way other HTML components are implemented.